internal/tcpchecker/checkport.go

```diff
@@ -0,0 +1,66 @@
+package tcpchecker
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"time"
+)
+
+func CheckPort(address string, port int, reporter Reporter) (string, error) {
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%v", address, port), 1*time.Second)
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Connect error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	err = conn.SetDeadline(time.Now().Add(1 * time.Second))
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Set deadline error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	scanner := bufio.NewScanner(conn)
+
+	output := ""
+	for scanner.Scan() {
+		output += fmt.Sprintf("%s\n", scanner.Text())
+	}
```

Comment:
This might be possible to simplify with a `strings.Builder` instead. The formatting and `.Text()` casting seems a but unusual to me, but dunno.

Reply:
Will try that 👍